11/debian-10/rootfs/opt/bitnami/scripts/libpostgresql.sh

```
@@ -710,7 +710,7 @@ postgresql_custom_init_scripts() {
 postgresql_stop() {
     if [[ -f "$POSTGRESQL_PID_FILE" ]]; then
         info "Stopping PostgreSQL..."
-        PGDATA="$POSTGRESQL_DATA_DIR" pg_ctl stop -w
+        PGDATA="$POSTGRESQL_DATA_DIR" pg_ctl stop -m $POSTGRESQL_SHUTDOWN_MODE -w
```

Take a look to my comment in `postgresql-env`

---

Hm, in that case I'd rather go for using the explicit invocation parameter (--timeout=<seconds>) instead of relying on an environment variable. Same actually applies to PGDATA (parameter would be --pgdata=<datadir>). As long as there is an explicit parameter available, that should be preferred over relying on environment variables, which are picked up or not picked up by a command (in contrast to a parameter, which is either supported or not and would throw an error at you, not silently ignore it).